pkg/domain/entities/types/images.go

```diff
@@ -147,6 +147,57 @@ type ImagePullReport struct {
 	ID string `json:"id,omitempty"`
 }
 
+// ImagePullReportV2 provides a detailed status of a container image pull operation.
+type ImagePullReportV2 struct {
+	// Status indicates the current state of the image pull operation.
+	// Possible values are:
+	// - "error":   An error occurred during the pull.
+	// - "pulling": The image pull is in progress.
+	// - "success": The image pull completed successfully.
+	Status string `json:"status,omitempty"`
+
+	// Images contains the unique identifiers of the successfully pulled images.
+	// This field is only populated when the Status is "success".
+	Images []string `json:"images,omitempty"`
+
+	// Stream provides a human-readable stream of events from the containers/image
+	// library. This data is intended for display purposes and should not be parsed
+	// by software.
+	Stream string `json:"stream,omitempty"`
+
+	// Progress provides detailed information about the download progress of an
+	// image layer. This field is only populated when the Status is "pulling".
+	Progress *ImagePullProgress `json:"progress,omitempty"`
+}
+
+// ImagePullProgress details the download progress of a single image layer.
+type ImagePullProgress struct {
+	// Current is the number of bytes of the current artifact that have been
+	// downloaded so far.
+	Current uint64 `json:"current,omitempty"`
+
+	// Total is the total size of the artifact in bytes. A value of -1
+	// indicates that the total size is unknown.
+	Total int64 `json:"total,omitempty"`
+
+	// Completed indicates whether the pull of the associated layer has finished.
+	// This is particularly useful for tracking the status of "partial pulls"
+	// where only a portion of the layers may be downloaded.
+	Completed bool `json:"completed,omitempty"`
+
+	// ProgressComponentID is the unique identifier for the artifact being downloaded.
+	// When the status is `PullComplete`, this field will contain the digest of the blobs at the source.
+	// If the status is `Success`, indicating that the entire PullOperation is complete,
+	// this field will contain the digest or ID of the artifact in local storage.
```

Comment:
- The whole of “progress” is documented to only be populated on “pulling”
- I think it’s _entirely unnecessary_, and confusing, to use a single field for two distinct values this way. AFAICT we can easily have two fields, each with a simpler documentatio.